src/Nethermind/Nethermind.Serialization.Rlp/NettyBufferMemoryOwner.cs

```diff
@@ -23,7 +23,8 @@ public NettyBufferMemoryOwner(IByteBuffer byteBuffer)
         _byteBuffer.Retain();
     }
 
-    ~NettyBufferMemoryOwner() {
+    ~NettyBufferMemoryOwner()
```

Comment:
As is Finaliser add to the regular `public void Dispose`

```
GC.SuppressFinalize(this);
```
As if the Dispose is directly called it doesn't need to call the finaliser? Will mean it will release the memory faster if Dispose is called

Reply:
Yes, tested to increase netty active bytes to 1.2GB instead of about 400MB if not explicitly called.

Reply:
Without the `GC.SuppressFinalize(this)` in `Dispose` when the GC comes to collect, it will put the `NettyBufferMemoryOwner` on the finalization queue to be collected later vs collecting it immediately. However since you've already called Dispose; when it runs the finalization queue `~NettyBufferMemoryOwner` will do nothing since its already disposed; so you might as well skip the additional delay in collection by suppressing the finalize (when the data has been disposed properly already)

If that makes sense?

Reply:
Yes